### PR TITLE
Publish Code Coverage task - Additional Files field Input validation for proper file path regex

### PR DESCRIPTION
**Pull request by abhishkk**
If user passes directory path instead of file path in additional files field, then directory path is ignored and not published to command.
Link to mseng [bug](https://mseng.visualstudio.com/VSOnline/_workitems/edit/1022209).

### CODE REVIEW

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts [hunk 1]
```diff
@@ -40,12 +40,13 @@ async function run() {
...
                         { matchBase: true });
-                    tl.debug(tl.loc('FoundNMatchesForPattern', additionalFileMatches.length, additionalFiles));
                 }
                 else {
                     // Use the specific additional file (no wildcards)
                     var additionalFileMatches: string[] = [additionalFiles];
                 }
+                additionalFileMatches = additionalFileMatches.filter(file => pathExistsAsFile(file));
+                tl.debug(tl.loc('FoundNMatchesForPattern', additionalFileMatches.length, additionalFiles));
             }
```
Comment (RanjanarMS) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: Spell check

Comment (RanjanarMS) [reply Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: What happens when no pattern match?

File: Tests-Legacy/L0/PublishCodeCoverageResults/publishCCResponses.json [hunk 2]
```diff
@@ -13,9 +13,27 @@
...
+    },
+    "stats": {
+        "some/path": {
+            "isFile": false
+        },
+        "some/path/one": {
+            "isFile": true
+        },
+        "some/path/two": {
+            "isFile": true
```
Comment (RanjanarMS) [inline Tests-Legacy/L0/PublishCodeCoverageResults/publishCCResponses.json]: Mixed mode where few are files and some are directory

Comment (abhishkk) [reply Tests-Legacy/L0/PublishCodeCoverageResults/publishCCResponses.json]: done

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts (hunk 1 above)
Comment (abhishkk) [reply Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: additionalcodecoveragefiles value is passed null to command. This is also shown in logs.

File: Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts [hunk 3]
```diff
@@ -97,4 +98,14 @@ function containsWildcard(inputValue: string) : boolean {
            inputValue.indexOf('?') >= 0;
 }
 
+// Gets whether the specified path exists as file.
+function pathExistsAsFile(path: string) {
+    try {
+        return tl.stats(path).isFile();
+    }
+    catch (error) {
+        return false;
```
Comment (nigurr) [inline Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: log the error in verbose atleast

Comment (abhishkk) [reply Tasks/PublishCodeCoverageResults/publishcodecoverageresults.ts]: done

**Pull request merged**